### ISSUE TITLE
nocache option not applying to recent points (repeat of #171)

### ISSUE THREAD

**Issue by jacksontj** [VictoriaMetrics/VictoriaMetrics#241]
**Describe the bug**
When a query with `nocache=1` is sent (e.g. `nocache=1&query=prometheus_build_info&time=1574709959.168`) the returned time has been modified (e.g. `1574709922`)

**To Reproduce**
Send a `nocache=1` query (as above) against a VM instance and note that the returned time won't match. Also note that if enough time has passed the query will start responding with the appropriate timestamp (presumably as the point falls out of memory -> disk)

**Expected behavior**
With `nocache=1` the returned time should match the queried time.

**Version**
The line returned when passing `--version` command line flag to binary. For example:
```
$ ./victoria-metrics-prod --version
victoria-metrics-20191121-211100-tags-v1.29.5-0-g3e9b7add
```

**Additional context**
Related to #171 (might be a regression). Upstream promxy issue: https://github.com/jacksontj/promxy/issues/244

**Comment by valyala** [comment]
Confirmed the issue. Will be fixed soon.

**Comment by valyala** [comment]
The issue should be fixed in the latest commit in both `master` and `cluster` branches.

**Comment by valyala** [comment]
The issue should be fixed in the release [v1.30.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing it.
